docs/limitations.md

```diff
@@ -0,0 +1,69 @@
+# Current Limitations and Installation Alternatives
+
+At least until [#3490](https://github.com/rectorphp/rector/issues/3490) is resolved, Rector relies on _runtime PHP reflection_ to analyze your source code. That means the code Rector works on cannot merely be treated as _data_, but has to be loaded (through the autoloader) into memory to be analyzed by the PHP interpreter.
+
+> Read the [PHPStan 0.12.26 release announcement](https://phpstan.org/blog/zero-config-analysis-with-static-reflection) to learn more about runtime and static reflection. PHPStan is the static analysis tool used by Rector.
+
+This brings a few limitations and affects your choices of how you can run Rector.
+
+## PHP Language Level
+
+The PHP process that runs Rector code also needs to load and parse your source code. So, you need to run a version of PHP that is both compatible with your project code as well as Rector.
+
+This means you cannot run Rector on PHP 8 to work on a project that uses language features limited to previous PHP versions.
+
+> One example for this is [array and string offset access using curly braces](https://www.php.net/manual/de/migration74.deprecated.php#migration74.deprecated.core.array-string-access-curly-brace).
+
+But also the other way round, you cannot run Rector on PHP 7 to analyze a project with PHP 8 language features, like [constructor property promotion](https://stitcher.io/blog/constructor-promotion-in-php-8).
```

Comment:
That should be doable, as promoted properties were added to php-parsed for PHP 7 too

Reply:
I'll just remove the remark then.

Reply:
I don't use Docker. Could you verify it in a GitHub Action?

Reply:
https://github.com/rectorphp/getrector.org/issues/269 contains an example of PHP 8 code that could not be processed with the PHP 7.3 Docker image, and so the Docker image was upgraded to use PHP 8.

Reply:
I see. I forgot to add this must be merged first, as code is now included manually.https://github.com/rectorphp/rector/pull/5665